ee/api/test/test_team_memberships.py

```diff
@@ -0,0 +1,380 @@
+
+        response = self.client.post(
+            "/api/projects/@current/explicit_members/",
+            {"user_uuid": new_user.uuid, "level": ExplicitTeamMembership.Level.ADMIN},
+        )
+        response_data = response.json()
+
+        self.assertDictEqual(
+            self.permission_denied_response("You don't have sufficient permissions in this project."), response_data
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_add_admin_as_project_admin_allowed(self):
+        self.organization_membership.level = OrganizationMembership.Level.ADMIN
+        self.organization_membership.save()
+        ExplicitTeamMembership.objects.create(
+            team=self.team, parent_membership=self.organization_membership, level=ExplicitTeamMembership.Level.ADMIN
+        )
+
+        new_user: User = User.objects.create_and_join(self.organization, "rookie@posthog.com", None)
+
+        response = self.client.post(
+            "/api/projects/@current/explicit_members/",
+            {"user_uuid": new_user.uuid, "level": ExplicitTeamMembership.Level.ADMIN},
+        )
+        response_data = response.json()
+
+        self.assertDictContainsSubset(
+            {"effective_level": ExplicitTeamMembership.Level.ADMIN, "level": ExplicitTeamMembership.Level.ADMIN,},
+            response_data,
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_add_member_to_non_current_project_allowed(self):
+        self.organization_membership.level = OrganizationMembership.Level.ADMIN
+        self.organization_membership.save()
+        another_team = Team.objects.create(organization=self.organization)
+
+        new_user: User = User.objects.create_and_join(self.organization, "rookie@posthog.com", None)
+
+        response = self.client.post(f"/api/projects/{another_team.id}/explicit_members/", {"user_uuid": new_user.uuid})
+        response_data = response.json()
+
+        self.assertDictContainsSubset(
+            {"effective_level": ExplicitTeamMembership.Level.MEMBER, "level": ExplicitTeamMembership.Level.MEMBER,},
+            response_data,
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_add_member_to_project_in_outside_organization_forbidden(self):
+        self.organization_membership.level = OrganizationMembership.Level.ADMIN
+        self.organization_membership.save()
+        _, new_team, new_user = User.objects.bootstrap("Acme", "mallory@acme.com", None)
+
+        response = self.client.post(f"/api/projects/{new_team.id}/explicit_members/", {"user_uuid": new_user.uuid,})
+        response_data = response.json()
+
+        self.assertDictEqual(self.not_found_response("Project not found."), response_data)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_add_member_to_nonexistent_project_forbidden(self):
```

Comment:
Another test case could be not being able to add a non-member of your org to a project to which you would have permissions.

Reply:
For sure